### PR TITLE
Feature/conan environ

### PR DESCRIPTION
**Pull request by lasote**
I'm not happy with the install method in manager. Now it creates in a tmp file a conanfile.txt file to emulate it when it receives a reference. It's necessary to process the reference as a depenency node in the graph because the generator needs the vars that the package reference declares.

About the rest, it's necessary to test virutalenv in OSx and Linux a lot, but basic usage is very good.
You can export this cmake recipe, should work in win but i've tested in linux:

https://github.com/lasote/conan-cmake-installer

### CODE REVIEW

File: conans/client/command.py [hunk 1]
```diff
@@ -299,6 +301,8 @@ def install(self, *args):
             reference = ConanFileReference.loads(args.reference)
         except:
             reference = os.path.normpath(os.path.join(current_path, args.reference))
+            if args.generator:
```
Comment (memsharded) [inline conans/client/command.py]: Probably not necessary, generators can be useful also when you have a conanfile.txt for many users, and for example one wants to install the YCM generator without modifying the conanfile.txt

File: conans/client/configure_environment.py [hunk 2]
```diff
@@ -1,22 +1,51 @@
...
-        self.os = getattr(self._settings, "os", None)
-        self.build_type = getattr(self._settings, "build_type", None)
-        self.libcxx = None
+        self._deps_env_info = deps_env_info
+        try:
+            self.compiler = str(self._settings.compiler)
+        except:
+            self.compiler = None
+
+        try:
```
Comment (memsharded) [inline conans/client/configure_environment.py]: why changing the previous oneliner to try-except?

File: conans/client/configure_environment.py [hunk 3]
```diff
@@ -58,7 +87,10 @@ def command_line(self):
             headers_flags = 'C_INCLUDE_PATH=%s CPP_INCLUDE_PATH=%s' % (include_paths, include_paths)
             command = "env %s %s %s %s %s" % (libs, ldflags, cflags, cpp_flags, headers_flags)
         elif self.os == "Windows" and self.compiler == "Visual Studio":
-            cl_args = " ".join(['/I"%s"' % lib for lib in self._deps_cpp_info.include_paths])
-            lib_paths = ";".join(['"%s"' % lib for lib in self._deps_cpp_info.lib_paths])
-            command = "SET LIB=%s;%%LIB%% && SET CL=%s" % (lib_paths, cl_args)
+                cl_args = " ".join(['/I"%s"' % lib for lib in self._deps_cpp_info.include_paths])
```
Comment (memsharded) [inline conans/client/configure_environment.py]: Seems double indentation here

File: conans/test/compile_helpers_test.py [hunk 4]
```diff
@@ -97,7 +97,16 @@ def configure_environment_test(self):
                                             '"path/to/includes/lib2"')
         # Not supported yet
         env = ConfigureEnvironment(BuildInfoMock(), MockWinGccSettings())
-        self.assertEquals(env.command_line, "")
+        self.assertEquals(env.command_line, 'env LIBS="-llib1 -llib2" LDFLAGS="-Lpath/to/lib1 '
```
Comment (memsharded) [inline conans/test/compile_helpers_test.py]: I don't understand this change, how is it possible that it was passing? these are the standard env flags, not really related to this feature

File: conans/client/configure_environment.py (hunk 2 above)
Comment (lasote) [inline conans/client/configure_environment.py]: It was not working. If setting is not defined it fails even with getattr

File: conans/test/compile_helpers_test.py (hunk 4 above)
Comment (lasote) [inline conans/test/compile_helpers_test.py]: It's not directly related but I added a this in the ConfigureEnvironment to generate the same configuration than linux gcc for mingw:  (self.os == "Windows" and self.compiler == "gcc"): (configure_environment.py line 58)

File: conans/client/configure_environment.py (hunk 2 above)
Comment (memsharded) [inline conans/client/configure_environment.py]: But how is it possible that it is not defined? You have an assert a few lines above

File: conans/client/loader.py [hunk 5]
```diff
@@ -216,3 +222,12 @@ def imports():
             for import_params in parameters:
                 conan_file.copy(*import_params)
         return imports
+
+    @staticmethod
```
Comment (memsharded) [inline conans/client/loader.py]: This is not used, can be removed

File: conans/client/configure_environment.py (hunk 2 above)
Comment (memsharded) [inline conans/client/configure_environment.py]: What about the assert?

**Comment by memsharded** [comment]
Can be merged, just check the unused code and the try-except that shouldn't be necessary, and merge

File: conans/client/configure_environment.py (hunk 2 above)
Comment (lasote) [inline conans/client/configure_environment.py]: I mean the "compiler" setting, not the setting object. If the recipe doesn't declare compiler as a setting it crashes.

**Pull request merged**